juriscraper/opinions/united_states/state/tex.py

```diff
@@ -14,298 +14,104 @@
-            )
-            self.status = 200
-            # self.take_screenshot()
-
-            nr_of_pages = self.find_element_by_xpath(
-                '//thead//*[contains(concat(" ", normalize-space(@class), " "), " rgInfoPart ")]/strong[2]'
-            )
-            records_nr = self.find_element_by_xpath(
-                '//thead//*[contains(concat(" ", normalize-space(@class), " "), " rgInfoPart ")]/strong[1]'
-            )
-            html_pages = []
-            if records_nr:
-                self.records_nr = int(records_nr.text)
-            if nr_of_pages:
-                if nr_of_pages.text == "1":
-                    text = self.webdriver.page_source
-                    self.webdriver.quit()
-
-                    html_tree = html.fromstring(text)
-                    html_tree.make_links_absolute(self.url)
-
-                    remove_anchors = lambda url: url.split("#")[0]
-                    html_tree.rewrite_links(remove_anchors)
-                    html_pages.append(html_tree)
-                else:
-                    logger.info(
-                        "Paginating through %s pages of results."
-                        % nr_of_pages.text
-                    )
-                    logger.info("  Getting page 1")
-                    text = self.webdriver.page_source
-
-                    html_tree = html.fromstring(text)
-                    html_tree.make_links_absolute(self.url)
-
-                    remove_anchors = lambda url: url.split("#")[0]
-                    html_tree.rewrite_links(remove_anchors)
-                    html_pages.append(html_tree)
-
-                    for i in range(int(nr_of_pages.text) - 1):
-                        logger.info(f"  Getting page {i + 2}")
-                        next_page = self.find_element_by_class_name(
-                            "rgPageNext"
-                        )
-                        next_page.click()
-                        self.webdriver.implicitly_wait(5)
-
-                        text = self.webdriver.page_source
-
-                        html_tree = html.fromstring(text)
-                        html_tree.make_links_absolute(self.url)
+        # The parameters required to filter in Texas
+        self.parameters = {
+            "ctl00$ContentPlaceHolder1$SearchType": "rbSearchByDocument",
+            "ctl00$ContentPlaceHolder1$dtDocumentFrom": str(last_month),
+            "ctl00$ContentPlaceHolder1$dtDocumentFrom$dateInput": last_month_str,
+            "ctl00_ContentPlaceHolder1_dtDocumentFrom_dateInput_ClientState": date_param,
+            "ctl00$ContentPlaceHolder1$dtDocumentTo": str(today),
+            "ctl00$ContentPlaceHolder1$dtDocumentTo$dateInput": today_str,
+            "ctl00$ContentPlaceHolder1$chkListDocTypes$0": "on",
+            "ctl00$ContentPlaceHolder1$btnSearchText": "Search",
+            "__VIEWSTATE": self.html.xpath("//input[@id='__VIEWSTATE']")[
+                0
+            ].get("value"),
```

Comment:
Similar to Ohio (I think?) it seems better not to do xpath stuff in here, but rather to pass it in. I do appreciate that you're doing date stuff in this method though instead of the init, unlike the other one.

Reply:
hmm... im not sure what is right here... what about just dropping the parsing into the parameters function

Reply:
Can you psuedo-code what you mean for me?

Reply:
ill just follow your suggestions in Ohio.

Reply:
Sounds good, I think. Assuming I said good things in Ohio.